File: TwoPointer/python/2473.py
n = int(input())
arr = sorted(map(int,input().split()))


def search(arr,n):
  idx_arr = [0 for _ in range(3)]
  min_sum = 10**18*9
  for i in range(n-2):
    left,right = i+1, n-1
    while left<right:
      total = arr[i] + arr[left] + arr[right]
      if (abs(total) < abs(min_sum)):
        idx_arr = [arr[i],arr[left],arr[right]]
        min_sum = total
      if total>0:
        right -= 1
      if total<0:
        left += 1
      if total == 0:
        return " ".join(map(str,sorted(idx_arr)))
  return " ".join(map(str,sorted(idx_arr)))

answer = search(arr,n)
print(answer)

File: TwoPointer/javascript/2473.js
const input = require("fs").readFileSync("/dev/stdin").toString().split("\n");

const n = Number(input[0]);
const arr = input[1]
  .split(" ")
  .map(Number)
  .sort((a, b) => a - b);

const search = (arr, n) => {
  let minSum = Infinity;

  const generateSum = (answer, num, idx) => {
    let [left, right] = [idx + 1, n - 1];
    if (minSum === 0) {
      return answer;
    }
    while (left < right) {
      const total = num + arr[left] + arr[right];
      if (Math.abs(minSum) > Math.abs(total)) {
        answer = [num, arr[left], arr[right]];
        minSum = total;
      }
      if (total < 0) {
        left += 1;
      }
      if (total > 0) {
        right -= 1;
      }
      if (total === 0) {
        return answer;
      }
    }
    return answer;
  };

  return arr.reduce(generateSum, [0, 0, 0]).join(" ");
};

const answer = search(arr, n);
console.log(answer);
